### ISSUE TITLE
VoiceOver: Accessibility issues in the "Toot an image" workflow

### ISSUE THREAD

**Issue by MarcoZehe** [DagAgren/toot#364]
When trying to toot an image with VoiceOver, there are several problems along the way which currently prevent blind users from tooting images with descriptions. While it might not be the most common thing for most, for some of us, sharing our experiences with sighted buddies also through pictures is important. Especially with the image recognition features built into iOS. So, the steps I roughly take are as follows, with the issues I encounter embedded.

### Start the toot

Starting the toot is accessible. You use the Toot button at the bottom of the screen. I write some text, maybe set a content warning, and the privacy.

### Attaching an image from the library

If I have taken a picture in the past and want to attach it, there are several problems:

1. The Attach from library button is not labeled properly, although the derived name makes it clear what the button does. Also see #363 for the unlabeled button.
2. Once double-tapped, the library opens to the Moments view.
   * There does not seem to be a way to change to a view of, say, all photos. Double-tapping the header does not change the view or give me a list of choices as I would expect, and there doesn't seem to be another button that would do that, either.
3. Each moments item should be a header. The label should get the AccessibilityHeader trait set so users can use the VoiceOver rotor to move between dates/moments.
4. Each item within a moment has a few problems:
   * Its label should be more descriptive, like the Photos app provides. Use something like "Photo" or "Video" at the start, then the date and time, possibly length (with videos). Look at, or listen to, what the Photos app does for inspiration.
   * It should have an AccessibilityImage trait so VoiceOver knows that it is dealing with photos and can apply its image recognition AI.
   * It should be checked that the AccessibilityFrame correctly encompasses each item. This corresponds to the VoiceOver-specific rectangle that gets drawn around the item VoiceOver is speaking.
   * Double-tapping to select an image currently does not work. If I double-tap an item that is spoken via its date, then find and double-tap the Done button, an image does not get attached to the toot, so I assume selecting it didn't work.
   * Once it does work, give each selected image the Selected trait, so VoiceOver automatically speaks the selected state and the user knows which image(s) will be attached. Obviously, clear that trait if the image gets deselected.

### Alternatively, attach an image from the camera

This works, it uses the iOS camera plug-in, with all the announcements VoiceOver provides like an automatic description of the image on some iPhones, the tilt, face recognition etc. That part is fully accessible out of the box.

### Editing an image

Once an image is attached, It gets both a labeled Remove button, and a "marker" item between the library and camera buttons. Double-tapping the latter gives an image editor to provide a description and crop the image. This one is currently largely inaccessible.

1. While I can add an image description via the text field, and end it with the Finish button on the keyboard, there is no way to dismiss the view *and* apply that description to the image. Turning VoiceOver off and tapping anywhere in the blank area that is the screen except the text field, dismisses the view, but the description gets thrown away. So:
   * Add proper Apply and Dismiss buttons so users can in a controlled fashion apply the description or discard the view.
2. While cropping isn't something most blind users would do, at least the photo area should be marked up correctly as an accessible item with an Image trait and a label consisting either of "image x" (a number of the image in the set), or the description that was already given to it by the user. It should also have the correct dimensions (AccessibilityFrame) so VoiceOver knows which area is the image.
3. As a bonus, as can be seen in the Photos app when editing a photo or cropping a video, make each handle an AccessibilityAdjustable, implementing the AccessibilityValue and AccessibilityIncrement and AccessibilityDecrement methods so users can actually adjust the frame of the image. Again, this might not be something most users who cannot see the image would do, but some might.

### Conclusion

With these changes, this big area of Toot will be accessible, and users can share their experiences just like their sighted peers can.

**Comment by DagAgren** [comment]
This should also now all be most fixed.
